src/main/java/core/basesyntax/utils/process/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.utils.process.impl;
+
+import core.basesyntax.storage.impl.StorageImpl;
+import core.basesyntax.utils.process.ShopService;
+import core.basesyntax.utils.transaction.FruitTransaction;
+import core.basesyntax.utils.transaction.handlers.OperationHandlerFactory;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationHandlerFactory factory;
+    private final StorageImpl storage;
+
+    public ShopServiceImpl(OperationHandlerFactory factory, StorageImpl storage) {
```

Comment:
The constructor parameter `StorageImpl storage` should be changed to use the interface type (e.g., `Storage storage`). This will make your code more flexible and easier to maintain.